reactos/drivers/usb/usbhub_new/usbhub.c

```diff
@@ -0,0 +1,5143 @@
+    {
+        KeSetEvent(&HubExtension->PendingRequestEvent,
+                   EVENT_INCREMENT,
+                   FALSE);
+    }
+
+    if (!InterlockedDecrement((PLONG)&HubExtension->ResetRequestCount))
+    {
+        KeSetEvent(&HubExtension->ResetEvent,
+                   EVENT_INCREMENT,
+                   FALSE);
+
+        if (HubExtension->HubFlags & USBHUB_FDO_FLAG_DEFER_CHECK_IDLE)
+        {
+            USBH_CheckHubIdle(HubExtension);
+        }
+    }
+}
+
+NTSTATUS
+NTAPI
+USBH_ChangeIndication(IN PDEVICE_OBJECT DeviceObject,
+                      IN PIRP Irp,
+                      IN PVOID Context)
+{
+    PUSBHUB_FDO_EXTENSION HubExtension;
+    USBD_STATUS UrbStatus;
+    BOOLEAN IsErrors = FALSE;
+    PUSBHUB_IO_WORK_ITEM HubWorkItem;
+    PUSBHUB_STATUS_CHANGE_CONTEXT HubWorkItemBuffer;
+    USHORT NumPorts;
+    USHORT Port;
+    NTSTATUS Status;
+    ULONG_PTR Bitmap;
+    ULONG BufferLength;
+
+    HubExtension = Context;
+    UrbStatus = HubExtension->SCEWorkerUrb.Hdr.Status;
+
+    DPRINT("USBH_ChangeIndication: IrpStatus - %p, UrbStatus - %p, HubFlags - %p\n",
+           Irp->IoStatus.Status,
+           UrbStatus,
+           HubExtension->HubFlags);
+
+    if (NT_ERROR(Irp->IoStatus.Status) || USBD_ERROR(UrbStatus) ||
+       (HubExtension->HubFlags & USBHUB_FDO_FLAG_DEVICE_FAILED) ||
+       (HubExtension->HubFlags & USBHUB_FDO_FLAG_DEVICE_STOPPING))
+    {
+        ++HubExtension->RequestErrors;
+
+        IsErrors = TRUE;
+
+        KeSetEvent(&HubExtension->StatusChangeEvent,
+                   EVENT_INCREMENT,
+                   FALSE);
+
+        if (HubExtension->HubFlags & USBHUB_FDO_FLAG_DEVICE_STOPPING ||
+            HubExtension->HubFlags & USBHUB_FDO_FLAG_DEVICE_FAILED ||
+            HubExtension->RequestErrors > 3 ||
+            Irp->IoStatus.Status == STATUS_DELETE_PENDING)
+        {
+            DPRINT("USBH_ChangeIndication: HubExtension->RequestErrors - %x\n",
+                   HubExtension->RequestErrors);
+
+            return STATUS_MORE_PROCESSING_REQUIRED;
+        }
+
+        DPRINT("USBH_ChangeIndication: HubExtension->RequestErrors - %x\n",
+               HubExtension->RequestErrors);
+    }
+    else
+    {
+        HubExtension->RequestErrors = 0;
+    }
+
+    BufferLength = sizeof(USBHUB_STATUS_CHANGE_CONTEXT) +
+                   HubExtension->SCEBitmapLength;
+
+    Status = USBH_AllocateWorkItem(HubExtension,
+                                   &HubWorkItem,
+                                   USBH_ChangeIndicationWorker,
+                                   BufferLength,
+                                   (PVOID *)&HubWorkItemBuffer,
+                                   DelayedWorkQueue);
+
+    if (!NT_SUCCESS(Status))
+    {
+        return STATUS_MORE_PROCESSING_REQUIRED;
+    }
+
+    RtlZeroMemory(HubWorkItemBuffer, BufferLength);
+
+    HubWorkItemBuffer->RequestErrors = 0;
+
+    if (IsErrors == TRUE)
+    {
+        HubWork
```

Comment:
Again a bug of GitHub. Starting with this message, the text is the same everywhere. Apparently this happens when there are too many messages.